Modules/System Tests/Cryptography Management/src/DESCryptoServiceProviderTest.Codeunit.al

```diff
@@ -0,0 +1,47 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+codeunit 132576 "Test DESCryptoServiceProvider"
+{
+    Subtype = Test;
+
+    [Test]
+    procedure TestEncryptText()
+    var
+        DESCryptoServiceProvider: Codeunit DESCryptoServiceProvider;
+        LibraryAssert: Codeunit "Library Assert";
+        EncryptedText: Text;
+        ExpectedEncryptedText: Text;
+    begin
+        // [GIVEN] With Encryption Key
+        ExpectedEncryptedText := GetCodDESCryptoServiceProviderEncryptedText();
+        // [WHEN] Encrypt Text 
+        EncryptedText := DESCryptoServiceProvider.EncryptTextWithDESCryptoServiceProvider('Test', 'Test1234', 'ABitofSalt');
+        // [THEN] Verify Result 
+        LibraryAssert.AreEqual(ExpectedEncryptedText, EncryptedText, 'Unexpected value when encrypting text using DESCryptoServiceProvider');
+    end;
+
+    procedure TestDecryptText()
+    var
+        DESCryptoServiceProvider: Codeunit DESCryptoServiceProvider;
+        LibraryAssert: Codeunit "Library Assert";
+        DecryptedText: Text;
+        ExpectedEncryptedText: Text;
+    begin
+        // [GIVEN] With Encryption Key
+        ExpectedEncryptedText := 'Test';
+        // [WHEN] Encrypt Text 
+        DecryptedText := DESCryptoServiceProvider.DecryptTextWithDESCryptoServiceProvider('Test', 'Test1234', 'ABitofSalt');
+        // [THEN] Verify Result 
+        LibraryAssert.AreEqual(ExpectedEncryptedText, DecryptedText, 'Unexpected value when encrypting text using DESCryptoServiceProvider');
+    end;
+
+
+    local procedure GetCodDESCryptoServiceProviderEncryptedText(): Text
+    begin
+        exit('·lá:Yy');
```

Comment:
Can we somehow get rid of this ('·lá:Yy') ?

Reply:
I tried to put in some other texts, but all encrypted texts give special characters. I will amend it that it checks on text length AND if it ends on "Yy", this should be safe enough :).